packages/desktop-client/src/components/autocomplete/CategoryAutocomplete.tsx

```diff
@@ -183,6 +208,39 @@ export function CategoryAutocomplete({
     [defaultCategoryGroups, categoryGroups, showSplitOption],
   );
 
+  const filterSuggestions = useMemo(
+    () =>
+      (
+        suggestions: (Suggestion | CategoryAutocompleteItem)[],
+        value: string,
+      ) => {
+        return suggestions
+          .filter((suggestion: Suggestion | CategoryAutocompleteItem) => {
+            const suggestionAsSuggestion = suggestion as Suggestion; // Cast to Suggestion
+            return (
+              suggestionAsSuggestion.id === 'split' ||
+              suggestionAsSuggestion.group?.name
+                .toLowerCase()
+                .includes(value.toLowerCase()) ||
+              (
+                suggestionAsSuggestion.group?.name +
+                ' ' +
+                suggestionAsSuggestion.name
+              )
+                .toLowerCase()
+                .includes(value.toLowerCase()) ||
+              defaultFilterSuggestion(suggestionAsSuggestion, value)
+            );
+          })
+          .sort(
+            (a, b) =>
+              customSort(a as Suggestion, value.toLowerCase()) -
+              customSort(b as Suggestion, value.toLowerCase()),
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏same type coercion issue here.